std/numeric.d

```diff
@@ -2572,27 +2572,55 @@ GapWeightedSimilarityIncremental!(R, F) gapWeightedSimilarityIncremental(R, F)
 
 /**
 Computes the greatest common divisor of $(D a) and $(D b) by using
-Euclid's algorithm.
+Euclid's or Stein's algorithm depending on compiler.
  */
 T gcd(T)(T a, T b)
 {
-    static if (is(T == const) || is(T == immutable))
-    {
-        return gcd!(Unqual!T)(a, b);
-    }
-    else
-    {
-        static if (T.min < 0)
+    version(DigitalMars) {
+        static if (is(T == const) || is(T == immutable))
         {
-            assert(a >= 0 && b >=0);
+            return gcd!(Unqual!T)(a, b);
+        }
+        else
+        {
+            static if (T.min < 0)
+            {
+                assert(a >= 0 && b >=0);
+            }
+            while (b)
+            {
+                immutable t = b;
+                b = a % b;
+                a = t;
+            }
+            return a;
         }
-        while (b)
+    } else {
+            static if (is(T == const) || is(T == immutable))
         {
-            immutable t = b;
-            b = a % b;
-            a = t;
+            return gcd!(Unqual!T)(a, b);
+        }
+        else {
+            import std.algorithm: swap;
+            import core.bitop: bsf;
+            if (a == 0)
+                return b;
+            if (b == 0)
+                return a;
+
+            immutable uint shift = bsf(a | b);
+            a >>= a.bsf;
+
+            do {
```

Comment:
> So I guess we want to normalize on { on its own line, too

Sure -> https://github.com/dlang/phobos/pull/4942

Also note that our "style" CI is already complaining at the moment :)